plugins/Compressor/CompressorControlDialog.cpp

```diff
@@ -497,10 +497,12 @@ void CompressorControlDialog::redrawKnee()
 	float actualRatio = m_controls->m_limiterModel.value() ? 0 : m_controls->m_effect->m_ratioVal;
 
 	// Calculate endpoints for the two straight lines
-	float kneePoint1 = m_controls->m_effect->m_thresholdVal - m_controls->m_effect->m_kneeVal;
-	float kneePoint2X = m_controls->m_effect->m_thresholdVal + m_controls->m_effect->m_kneeVal;
-	float kneePoint2Y = (m_controls->m_effect->m_thresholdVal + (-m_controls->m_effect->m_thresholdVal * (actualRatio * (m_controls->m_effect->m_kneeVal / -m_controls->m_effect->m_thresholdVal))));
-	float ratioPoint = m_controls->m_effect->m_thresholdVal + (-m_controls->m_effect->m_thresholdVal * actualRatio);
+	const float thresholdVal = m_controls->m_effect->m_thresholdVal;
+	const float kneeVal = m_controls->m_effect->m_kneeVal;
+	float kneePoint1 = thresholdVal - kneeVal;
+	float kneePoint2X = thresholdVal + kneeVal;
+	float kneePoint2Y = (thresholdVal != 0.f) ? (thresholdVal + (-thresholdVal * (actualRatio * (kneeVal / -thresholdVal)))) : 0.f;
```

Comment:
Could you cancel `-thresholdVal` from the product in here to get `thresholdVal + actualRatio * kneeValue`? Then you don't need to check the threshold, and the expression will be continuous at zero threshold.

Reply:
Good catch.  I thought something seemed odd about that code, but I didn't bother looking into it because I assumed me four years ago knew what he was doing.  But me four years ago never knew what he was doing.